lib/util/version.js

```diff
@@ -21,14 +21,33 @@ function resetDetectedVersion() {
   cachedDetectedReactVersion = undefined;
 }
 
+function resolvePkgPath(pkg, context) {
+  const filepath = context && context.getFilename();
+  const basedir = filepath ? path.dirname(filepath) : process.cwd();
+  try {
+    return resolve.sync(pkg, {basedir});
+  } catch (err) {
+    // https://github.com/eslint/eslint/issues/11989
+    if (err.code !== 'INVALID_BASEDIR' || !/[/\\]\d+_[^/\\]*\.[\da-z]+$/i.test(filepath)) {
```

Comment:
i'm really not confident about this regex. why do we even need to do the `resolve` call if there's a virtual filename?

Reply:
`{index}_` prefix is hard coded at https://github.com/eslint/eslint/blob/master/lib/linter/linter.js#L1303

Reply:
so it kind of seems like, if `path.dirname()` exists and is a file, and if the file does *not* exist, then it's a virtual filename?

Reply:
@ljharb 

If `path.dirname()` exists and is a file, then the file can never exist, because dirname can not be file and dir at the same time.

I just want to ensure it is a virtual filename generated by ESLint itself specifically instead of passing accidentally by call this package's API.

If you think a loose test is fine, the regexp test can be removed, change to test dirname directly.

Reply:
Given that eslint itself is being very loose by providing this virtual filename, i'd rather rely on fs methods to detect when this is happening. Nobody is likely to be using this API but us, so I'm not worried about constraining it.

Reply:
OK, I'll simply the implementation.